### PR TITLE
Add first stab at arch docs.

### PR DESCRIPTION
**Pull request by nishkrishnan**
We can keep adding to this as we find more things to do.

### CODE REVIEW

File: server/neptune/README.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+Plan review signals are received directly by this workflow from gateway which pulls the workflow ID from the check run's `External ID` field.  
+
+If the plan is approved, the workflow proceeds with the apply, all the while updating the parent execution with the status, before exiting the workflow.
+If the plan is rejected or times out (1 week timeout on plan reviews), the parent is notified and the workflow exits.
+
+### Retries
+
+The workflow itself has no retries configured.  All activities use the default retry policy except for Terraform Activities.  Terraform Activities throw up a `TerraformClientError` if there is an error from the binary itself.  This error is configured to be non-retryable since most of the time this is a user error.  
+
+For Terraform Applies, timeouts are not retried. Tiemouts can happen from exceeding the ScheduleToClose threshold or from lack of heartbeat for over a minute.  Instead of retrying the apply, which can have unpredictable results, we signal our parent that there has been a timeout and this is surfaced to the user.
```
Comment (samrabelachew) [inline server/neptune/README.md:150]: nit: timeouts

File: server/neptune/README.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+* Start/signal the deploy temporal workflow for each root that requires processing.
+
+#### Pull
+* Clone repository on disk and fetch root configuration.
+* Get the latest diff of the entire PR and determine if any roots have changed.
+* Proxy event to legacy worker if any roots require processing.
+
+#### Check Run
+* Determine which action is being triggered.  We have custom buttons that are thrown up on the check run depending on the situation.  As of now there are 2 types of events we are looking for.  `unlocked` and `plan_review`.  When we receive both these events, we signal our deploy workflow and terraform workflow respectively.
+
```
Comment (samrabelachew) [inline server/neptune/README.md:39]: Add section reruns/checksuite? Maybe something like:

```suggestion
#### Check Run
* Determine which action is being triggered.  We have custom buttons that are thrown up on the check run depending on the situation.  As of now there are 2 types of events we are looking for:  `unlocked` and `plan_review`.  When we receive both these events, we signal our deploy workflow and terraform workflow respectively.
* Listens for the GH provided `Re-run failed checkruns` button selection and signals our deploy workflow that we are attempting to add the previously failing revision back into the deploy queue for a rerun attempt. This is similar to the check suite event described below, but the key difference here is this request only reruns attempts where a checkrun has failed. 

#### Check Suite
* Listens for GH provided `Re-run all checkruns` button selection events and signals our deploy workflows that we are attempting add all off the modified roots within a revision back into their respective deploy queues for a rerun attempt, regardless of success or failure status. Note that we only support rerun attempts if a revision meets the following criteria:
  - check run request comes from a revision on the default branch of a repo (force applies are not allowed)
  - revision is identical to the latest persisted deployment attempt for a specific root
```

File: server/neptune/README.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+State changes are reflected in the github checks UI (ie. plan in progress, plan failed, apply in progress etc.).  A single check run is used to represent the deployment state.  The check run state is indicative of the completion state of the deployment and the details of the deployment itself are rendered in the check run details section.
+
+State changes for apply jobs specifically are sent to SNS for internal auditing purposes.
+
+## Terraform
+
+The Terraform workflow runs on the granularity of a single deployment.  It's identifier is the deployment's identifier which is randomly generated in the Deploy Workflow.  Note: this means a single revision can be tied to multiple deployments.
+
+The terraform workflow is stateful due to the fact that it keeps data on disk and references it throughout the workflow.  Cleanup of that data only happens when that workflow is complete.  In order to ensure this statefulness, The terraform workflow is aware of the worker it's running on and fetches this information as part of the first activity.  Each successive activity task takes place on the same task queue.
```
Comment (Aayyush) [inline server/neptune/README.md:137]:
```suggestion
The terraform workflow is stateful due to the fact that it keeps data on disk and references it throughout the workflow.  Cleanup of that data only happens when that workflow is complete.  In order to ensure this statefulness, the terraform workflow is aware of the worker it's running on and fetches this information as part of the first activity.  Each successive activity task takes place on the same task queue.  
```

**Review by Aayyush — DISMISSED** [review]
Was a good overall architecture refresher for me as well  :) And the arch diagram looks really good 💯

**Review by samrabelachew — APPROVED** [review]

**Pull request merged**